src/main/java/org/apache/sysds/runtime/instructions/fed/MMChainFEDInstruction.java

```diff
@@ -43,6 +44,12 @@ public MMChainFEDInstruction(CPOperand in1, CPOperand in2, CPOperand in3,
 		_type = type;
 	}
 	
+	public MMChainFEDInstruction(MMChainCPInstruction instruction) {
+		this(instruction.input1, instruction.input2, instruction.input3, instruction.output,
+			instruction.getMMChainType(), -1 /* ignore numThreads */, instruction.getOpcode(),
+			instruction.getInstructionString());
```

Comment:
I will update this to use the same `numThreads`, but I don't think this has any effect for the federated implementation.